src/zcbenchmarks.cpp

```diff
@@ -31,6 +31,39 @@
 #include "zcash/IncrementalMerkleTree.hpp"
 
 using namespace libzcash;
+// This method is based on Shutdown from init.cpp
+void shutdown_wallet()
+{
+    LogPrintf("%s: In progress...\n", __func__);
+    if (ShutdownRequested())
+        throw new std::runtime_error("The node is shutting down");
+
+    if (pwalletMain)
+        pwalletMain->Flush(false);
+#ifdef ENABLE_MINING
+    GenerateBitcoins(false, NULL, 0);
+#endif
+    UnregisterNodeSignals(GetNodeSignals());
+    if (pwalletMain)
+        pwalletMain->Flush(true);
+
+    UnregisterAllValidationInterfaces(); 
+    delete pwalletMain;
+    pwalletMain = NULL;
+    bitdb.Reset();
+    LogPrintf("%s: done\n", __func__);
+}
+
+void restart_wallet_after_load(){
+    RegisterNodeSignals(GetNodeSignals());
```

Comment:
In `init.cpp`, this happens before wallet loading. So we should move this into `shutdown_wallet()` just after `bitdb.Reset()`, at which point I'd also rename `shutdown_wallet()` to `pre_wallet_load()`, and rename `restart_wallet_after_load()` to `post_wallet_load()`.

Reply:
Umm.. so should we do it in both? Otherwise will the same state be recovered after the benchmark?

Reply:
No, it should be moved. This is why I suggested renaming the functions: they shouldn't correspond to "shutdown" and "restart", but instead "restart process up to just before wallet load" and "restart process from just after wallet load".

Reply:
What does that method do exactly? Is it clear that we need to do this unregistering and reregistering at all.  The unregistering is part of `Shutdown` but maybe not important for unloading the wallet?

Reply:
It looks like it disconnects and reconnects the core node with the network subsystem. I would assume that's why there is a flush before and after on disconnect, so that all captured network activity is flushed and no new activity interferes with the wallet during shutdown. As to whether it is necessary in this instance: probably not in regtest mode, but very likely in other nodes, so probably best to maintain it? Other review comments welcome.

Reply:
updated